### ISSUE TITLE
Xiaomi Philips Eyecare 2nd Generation

### ISSUE THREAD

**Issue by misterymisery** [syssi/philipslight#25]
After few hours (let say they stop to work overnight) touch buttons on lamp are locked and only lamp restart fix this. After restart they are good again for few hours. Any way to fix this?

**Comment by seafoodrice** [comment]
Seems this issue has been dragged on for long and no one able to proceed with more diagnosis and fixes. Anyone can help one this as this breaks the core feature to support this the device.

**Comment by bazuchan** [comment]
I have same problem with xiaomi zhimi heater and it started after firmware update.

**Comment by Stupco** [comment]
Same issue with 2x Philips Eyecare lamps, only happened since integrating with HA which is a shame as they are now useless for "dumb" operations (using buttons) which my wife prefers.

Would really like a fix!

Light Xiaomi firmware version: 1.2.8.0026

**Comment by SpeedFire0** [comment]
> After few hours (let say they stop to work overnight) touch buttons on lamp are locked and only lamp restart fix this. After restart they are good again for few hours. Any way to fix this?
> 
> PS.
> I have 3 of them and all behave same. When removing them from HA and deleting from custom_componets they are working fine.

I had the same issue after implementing this component. However all was fixed after I plugged out and plugged in the lamp into the power socket. Both hardware and software controls restored completely. You can try.

**Comment by pniewiadowski** [comment]
> > After few hours (let say they stop to work overnight) touch buttons on lamp are locked and only lamp restart fix this. After restart they are good again for few hours. Any way to fix this?
> > PS.
> > I have 3 of them and all behave same. When removing them from HA and deleting from custom_componets they are working fine.
> 
> I had the same issue after implementing this component. However all was fixed after I plugged out and plugged in the lamp into the power socket. Both hardware and software controls restored completely. You can try.

That's work for some time and then happen again :/

**Comment by Stupco** [comment]
Confirming this bug is still an issue.

**Comment by syssi** [comment]
Please subscribe this issue too: https://github.com/rytilahti/python-miio/issues/617

**Comment by syssi** [comment]
Does this issue apply to `philips.light.sread1` and `philips.light.sread2`?

**Comment by SpeedFire0** [comment]
> Does this issue apply to `philips.light.sread1` and `philips.light.sread2`?

philips.light.sread2 works absolutely fine, while physical buttons on philips.light.sread1 stop working after several days (then you need to re-plug it to the power socket)

**Comment by laoyang945** [comment]
I have the same issue with `philips.light.sread2`, but I cannot 100% reproduce it.

**Comment by xekil** [comment]
same problem, help please !

**Comment by samoswall** [comment]
The same problem. Is there a way to fix it?
`philips.light.sread1`  Firmware version 2.1.3.0024 latest/

**Comment by Stupco** [comment]
Yup, definitely still an issue. It's really pissing off my wife, basically the lamps are unusable as smart lamps, might need to just make them dumb lamps until this is fixed...

**Comment by pniewiadowski** [comment]
I switch to this integration some time ago and it doens't have this issue, so here is my recomendation:
https://github.com/syssi/xiaomi_raw
Example switches.yaml configuration:
```
- platform: xiaomi_miio_raw
  name: przemek_desk_light
  host: 10.10.10.13
  token: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
  turn_on_command: 'set_power'
  turn_on_parameters: 'on'
  turn_off_command: 'set_power'
  turn_off_parameters: 'off'
  state_property: 'power'
  state_property_getter: 'get_prop'
  state_on_value: 'on'
  state_off_value: 'off'    
```

**Comment by samoswall** [comment]
Configured **miio** integration.
![2021-09-22_10-26-42](https://user-images.githubusercontent.com/80251553/134301400-77fc9c11-ba6e-4ccf-bfa6-73b512ed640e.png)
I'm testing it (waiting for a few days).

**Comment by misterymisery** [comment]
> I switch to this integration some time ago and it doens't have this issue, so here is my recomendation:
> https://github.com/syssi/xiaomi_raw
> Example switches.yaml configuration:
> 
> ```
> - platform: xiaomi_miio_raw
>   name: przemek_desk_light
>   host: 10.10.10.13
>   token: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
>   turn_on_command: 'set_power'
>   turn_on_parameters: 'on'
>   turn_off_command: 'set_power'
>   turn_off_parameters: 'off'
>   state_property: 'power'
>   state_property_getter: 'get_prop'
>   state_on_value: 'on'
>   state_off_value: 'off'    
> ```

That is great. It is working few hours flawlessly for now where sometimes it started to fail after 2 hours! I can't wait if this still work tomorrow. Now how to fix it in offcial release?!

**Comment by syssi** [comment]
Did you see the discussion here? https://github.com/rytilahti/python-miio/issues/617

As soon a working / proper / issue free set of properties is known this component can be fixed.

**Comment by misterymisery** [comment]
@syssi yeap I seen it. Already responded there but not sure who owns this fix? Is it you?

**Comment by xekil** [comment]
Any ideas on the availability of the fix for the physical buttons problem?
thank you

**Comment by xm4rcell0x** [comment]
Any news? The bug is still here

**Comment by xekil** [comment]
Any news, is there a viable control solution? Any feedback? Thanks
